driver-core/src/main/com/mongodb/internal/operation/AsyncCommandBatchCursor.java

```diff
@@ -193,39 +191,44 @@ public int getMaxWireVersion() {
 
     private void getMore(final ServerCursor cursor, final SingleResultCallback<List<T>> callback) {
         resourceManager.executeWithConnection((connection, wrappedCallback) ->
-                        assertNotNull(connection).commandAsync(namespace.getDatabaseName(),
-                                getMoreCommandDocument(cursor.getId(), connection.getDescription(), namespace,
-                                        limit, batchSize, count.get(), maxTimeMS, comment),
-                                NO_OP_FIELD_NAME_VALIDATOR, ReadPreference.primary(),
-                                CommandResultDocumentCodec.create(decoder, NEXT_BATCH),
-                                assertNotNull(resourceManager.getConnectionSource()).getOperationContext(),
-                                (commandResult, t) -> {
-                                    if (t != null) {
-                                        Throwable translatedException =
-                                                t instanceof MongoCommandException
-                                                        ? translateCommandException((MongoCommandException) t, cursor)
-                                                        : t;
-                                        wrappedCallback.onResult(null, translatedException);
-                                        return;
-                                    }
-                                    CommandCursorResult<T> commandCursorResult = toCommandCursorResult(
-                                            connection.getDescription().getServerAddress(), NEXT_BATCH, assertNotNull(commandResult));
-                                    resourceManager.setServerCursor(commandCursorResult.getServerCursor());
-
-                                    if (!resourceManager.operable()) {
-                                        // The cursor was closed
-                                        resourceManager.releaseServerAndClientResources(connection);
-                                        wrappedCallback.onResult(emptyList(), null);
-                                        return;
-                                    }
-
-                                    List<T> nextBatch = commandCursorResult.getResults();
-                                    if (nextBatch.isEmpty() && commandCursorResult.getServerCursor() != null) {
-                                        getMore(commandCursorResult.getServerCursor(), wrappedCallback);
-                                    } else {
-                                        wrappedCallback.onResult(nextBatch, null);
-                                    }
-                                }), callback);
+                getMoreLoop(assertNotNull(connection), cursor, wrappedCallback), callback);
+    }
+
+    private void getMoreLoop(final AsyncConnection connection, final ServerCursor serverCursor,
```

Comment:
I wonder if using `AsyncCallbackLoop` will make the code simpler (it will remove the `getMoreLoop` recursion, but that does not necessary mean that the code will be simpler). I will play with the code and see how it is.

Reply:
No, it hardly reduces the amount of code, and makes producing a result from a loop more difficult (requires additional code to do that), so the change is not beneficial here.